pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/RuntimeSpawner.java

```diff
@@ -139,8 +139,30 @@ public void join() throws Exception {
         });
     }
 
+    /**
+     * Terminates the runtime
+     */
     @Override
     public void close() {
+        // cancel liveness checker before terminating runtime.
```

Comment:
I am not sure the change code here is related to Authentication. If there are multiple changes coupled in this pull request, can we try to separate them into multiple different pull requests in future? or at least highlight them at the description?

Reply:
This was a change that was required in a previous version of the implementation. I don't think its required anymore